### PR TITLE
chore: refactor notification with a wrapper

### PR DESCRIPTION
**Pull request by nicolasburtey**

### CODE REVIEW

File: app/navigation/root-navigator.tsx [hunk 1]
```diff
@@ -271,18 +270,16 @@ export const RootStack: NavigatorType = () => {
       })
   }, [])
 
+  useAddDeviceToken()
```
Comment (nicolasburtey) [inline app/navigation/root-navigator.tsx:273]: not needed?

File: app/navigation/root-navigator.tsx [hunk 2]
```diff
@@ -271,18 +270,16 @@ export const RootStack: NavigatorType = () => {
...
     const setupNotifications = async () => {
-      if (hasToken && client) {
-        const hasPermission = await hasNotificationPermission()
-        if (hasPermission) {
-          addDeviceToken(client)
-          messaging().onTokenRefresh(() => addDeviceToken(client))
-        }
+      const hasPermission = await hasNotificationPermission()
+      if (hasPermission) {
+        messaging().onTokenRefresh(useAddDeviceToken)
```
Comment (UncleSamtoshi) [inline app/navigation/root-navigator.tsx:278]: Hooks need to run on every render, I dont believe they can be used in a useEffect which happens outside of a render cycle and only when its dependencies change.

File: app/navigation/root-navigator.tsx (hunk 1 above)
Comment (UncleSamtoshi) [inline app/navigation/root-navigator.tsx:273]: I believe this will call addDeviceToken on every render.

**Review by UncleSamtoshi — CHANGES_REQUESTED** [review]

**Comment by daviroo** [comment]
@nicolasburtey this needs rebased

**Pull request closed without merging**